ua-snap/data-api
# New demographics fields + improved CSV

This PR fetches the new fields added to the demographics data [during this PR](https://github.com/ua-snap/epa-justice/pull/15), limits API returns to communities with >=50 people,  and improves the CSV output by tranposing the table, adding more detailed descrioptions of the data elements, and re-ordering the fields to match the display order in NCR. 

closes #490 
closes #489 
closes #487 
closes #482
closes #476 
closes #475 

To test:
- set the `API_GS_BASE_URL` environmental variable to the development server like so:
```export API_GS_BASE_URL=https://gs.earthmaps.io/geoserver/```
- fire up the API and request demographics data (including CSVs) for some communities. Communities with <50 people should return a basic JSON error message. For example, try Fairbanks and Nabesna.
```
# Fairbanks should return data as JSON and CSV
http://127.0.0.1:5000/demographics/AK124
http://127.0.0.1:5000/demographics/AK124?format=csv

# Nabesna should not return data as JSON or CSV
http://127.0.0.1:5000/demographics/AK256
http://127.0.0.1:5000/demographics/AK256?format=csv
```

The descriptive text in the CSV output has grown considerably here, and so could use some careful proofreading to make sure the sentences are sane and reference the actual datasets that the particular line of data came from. Crosschecking with the `README.md` would be much appreciated!

**Note:** In order to use `gs.earthmaps.io` in this branch, I had to bypass SSL certificate verification in the route-specific `demographics.validate_community_id()` function and also in more widely used `fetch_data.make_get_request()` function. I didn't know how else to get this to work, but I imagine we need to change this in production. Any thoughts on how & when is best to re-introduce the SSL verification would be welcomed!
